testutils/inject/analog.go

```diff
@@ -30,3 +32,21 @@ func (a *Analog) ReadCap() []interface{} {
 	defer func() { a.readCap = nil }()
 	return a.readCap
 }
+
+// Write calls the injected Write or the real version.
+func (a *Analog) Write(ctx context.Context, value int, extra map[string]interface{}) error {
+	a.readCap = []interface{}{ctx}
```

Comment:
Looking at the documentation you left for `WriteCap`, I suspect we should also store the value here. 

...though TBH I don't understand when you'd ever need to call `WriteCap()` or `ReadCap()` and don't really get why this code exists at all. The one use of `ReadCap` in our codebase is used to assert that the context passed into a function really is the context passed in, which seems like kind of a useless test.

Reply:
I'm not going to change the pattern of inject implementations as part of this pr. They could probably be better, but they do their job of injecting interfaces for testing, and I'm happy with that.

Reply:
I did not mean to change the pattern; I don't think you're following the pre-existing pattern here. For example, in testutils/inject/gpio_pin.go, the captures store all of the arguments except `extra`, which would include storing `value` here.

I wonder if there's some other reason not to store `value` and to set things up the way you did, but I don't see that reason myself, partly because I don't understand why the pattern has been set up this way in the first place.

Reply:
Ah I see, no you're right, we should store value here. tbh I do think our injects could be better. The `Cap` functions are usually to capture the input arguments to the injected function to be checked against in server-client code tests. But to be frank our team almost never has a need to use the Cap functions in testing the drivers, because we care about testing the behaviour of inputs not validating that we've sent the inputs. The `Cap` functions are usually used in server tests for resource APIs.

... the inject package could be better and lessconfusing overall, maybe one day.